packages/client/src/LensClient.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { Wallet } from 'ethers';
+
+import LensClient, { Credentials, mumbai, ReactionTypes } from '.';
+
+let credentials: Credentials;
+const getCredentials = async (wallet: Wallet) => {
+  if (credentials) {
+    return credentials;
+  }
+  const client = LensClient.init(mumbai);
+  const address = await wallet.getAddress();
+  const challenge = await client.generateChallenge(address);
+  const signature = await wallet.signMessage(challenge);
+
+  credentials = await client.authenticate(address, signature);
+  return credentials;
+};
+
+const getAuthenticatedClient = async (wallet: Wallet) => {
```

Comment:
nit: I would suggest traditional function declaration for consistency with the rest of the codebase.